src/main/java/teammates/logic/api/Logic.java

```diff
@@ -754,6 +755,40 @@ public void deleteStudentsInCourseCascade(String courseId) {
         studentsLogic.deleteStudentsInCourseCascade(courseId);
     }
 
+    public void deleteStudentsInCourseParallel(String courseId) {
```

Comment:
I feel like fetching the whole object just to delete it is unnecessary. We can just get the id

Reply:
I agree. Is it possible to do key only query to get the id and proceed to delete by id?

Reply:
Updated the functions to perform a query to only retrieve the `id` for Student, FeedbackResponse and FeeedbackResponseComment (`email` and `courseId` for Student) using `project` method from the objectify [package](https://www.javadoc.io/static/com.googlecode.objectify/objectify/6.0/com/googlecode/objectify/cmd/Query.html#project-java.lang.String...-).